### PR TITLE
[kfp sdk] Added examples for ArtifactLocation, ResourceOp, VolumeOp, and Sidecar.

### PR DESCRIPTION
**Pull request by eterna2**
Recent PRs have added quite a few new features to kfp sdk, so I have updated the `samples/basic` with the following examples:
- `artifact_location.py` on how to config artifact location in the pipeline workflow template
- `sidecar.py` on how to add sidecars to containerOp in a pipeline

> ### NOTE
> As requested by @hongye-sun in #1064, I have added an example for my PR. But I've also added a
> few more example for recent PR too.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1338)

### CODE REVIEW

**Comment by elikatsis** [comment]
These `ResourceOp` and `VolumeOp` samples are also under `samples/resourceops/`, along with other samples on `VolumeOp`s and `VolumeSnapshotOp`s.
Maybe it's better to rearrange those, e.g. create the directories `samples/volumeops/` and `samples/volumesnapshotops` or something like that.

**Comment by eterna2** [comment]
Ah, I didn't realize that. Must have missed them.

I will just remove them and just leave artifact location and sidecar.

**Comment by elikatsis** [comment]
>I will just remove them and just leave artifact location and sidecar.

Sure, I'll make another PR rearranging them then. Thanks!

File: samples/basic/artifact_location.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        insecure=False,
+    )
+
+    # set pipeline level artifact location
+    dsl.get_pipeline_conf().set_artifact_location(pipeline_artifact_location)
+
+    # pipeline level artifact location (to minio)
+    op1 = dsl.ContainerOp(name="foo", image="busybox:%s" % tag)
+
+    # op level artifact location (to s3 bucket)
```
Comment (Ark-kun) [inline samples/basic/artifact_location.py]: Can we not promote the op-level artifact location for now?

Comment (eterna2) [reply samples/basic/artifact_location.py]: sure. updated.

**Comment by eterna2** [comment]
@Ark-kun updated and remove op-level conf.

**Pull request merged**